### PR TITLE
Assignment 4 donde

### PR DESCRIPTION
**Pull request by ssolana**
I couldn't upload the assignment 4 files into a new pull request, because my first pull request from assignment2 hadn't been accepted, in order to not losing the assignment2 files i have to upload them all with the assignment4 files, I'm sorry i couldn't find any other option

### CODE REVIEW

**Comment by ocorcho** [comment]
Assignment 2a and 2b are now ok

File: Assignment4/ssolana-a180262/task06.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1K6X2wBziy_YbFT5yAM4JDq_x7_ZjnvKZ
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/ssolana-a180262/task06.py:12]: This line has to be commented out in order for the script to work

File: Assignment4/ssolana-a180262/task07.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s, "is a subclass of Person")
+  for s1, p1, o1 in g.triples((None, RDFS.subClassOf, s)):
+    print(s1, "is a subclass of", s, "and of Person")
+
+print("SPARQL:")
+from rdflib.plugins.sparql import prepareQuery
+q1 = prepareQuery ('''
+  SELECT ?Subject WHERE{
+    ?Subject(rdfs:subClassOf/rdfs:subClassOf*) ns:Person
```
Comment (dgarijo) [inline Assignment4/ssolana-a180262/task07.py:39]: This is equivalent to just having rdfs:subClassOf+

File: Assignment4/ssolana-a180262/task07.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+q1 = prepareQuery('''
+    SELECT DISTINCT ?Subject ?p ?o WHERE {
+      {
+        ?Subject rdf:type ?Subject2 .
+        ?Subject2 (rdfs:subClassOf/rdfs:subClassOf*) ns:Person .
+        ?Subject ?p ?o
+      } UNION
+      {
+        ?Subject3 rdf:type ns:Person .
+        ?Subject ?p ?o
```
Comment (dgarijo) [inline Assignment4/ssolana-a180262/task07.py:105]: This is wrong. It should be ?subject3, or otherwise you are binding all triples in the graph

**Comment by dgarijo** [comment]
Please fix the main typo in the last query

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please see my comments to address current issues

**Comment by dgarijo** [comment]
Requested changes have not been implemented. Closing this PR

**Pull request closed without merging**